### PR TITLE
[SYCL][COMPAT] Added helper functions to manage library and kernel rt loads

### PR DESCRIPTION
**Pull request by Alcpz**
Marked as draft until #13027 gets merged.

- Adds functionality to SYCLcompat to load and unload libraries at runtime.
- Adds helper functions to load kernel functors from said libraries.

Minor:
- Tests updated to match the style of the rest of the library

### CODE REVIEW

File: sycl/doc/syclcompat/README.md [hunk 1]
```diff
@@ -1177,16 +1177,48 @@ kernel names during machine translation.
...
 SYCL spec supported by the current SYCL compiler.
 
+The `SYCLCOMPAT_CHECK_ERROR` macro encapsulates an error-handling mechanism for
+expressions that might throw `sycl::exception` and `std::runtime_error`.
+If no exceptions are thrown, it returns `syclcompat::error_code::SUCCESS`.
+If a `sycl::exception` is caught, it returns `syclcompat::error_code::BACKEND_ERROR`.
+If a `std::runtime_error` exception is caught,
+`syclcompat::error_code::DEFAULT_ERROR` is returned instead.
+For both cases, it prints the error message to the standar
+d error stream.
```
Comment (AidanBeltonS) [inline sycl/doc/syclcompat/README.md:1187]:
```suggestion
For both cases, it prints the error message to the standard
error stream.
```

File: sycl/doc/syclcompat/README.md [hunk 2]
```diff
@@ -1177,16 +1177,48 @@ kernel names during machine translation.
...
+#ifdef _WIN32
+#define SYCLCOMPAT_EXPORT __declspec(dllexport)
+#else
+#define SYCLCOMPAT_EXPORT
+#endif
 
-template <class... Args> class sycl_compat_kernel_name;
-template <int Arg> class sycl_compat_kernel_scalar;
+namespace syclcompat {
+enum error_code { SUCCESS = 0, DEFAULT_ERROR = 999 };
```
Comment (AidanBeltonS) [inline sycl/doc/syclcompat/README.md:1218]: Missing `BACKEND_ERROR`

**Review by AidanBeltonS — APPROVED** [review]

File: sycl/doc/syclcompat/README.md (hunk 2 above)
Comment (Alcpz) [reply sycl/doc/syclcompat/README.md]: These were conflicting due to your previous review. Sorry about that, it's been fixed once the merge conflicts were resolved. Thanks!

**Comment by Alcpz** [comment]
@intel/llvm-gatekeepers, please merge. Thanks!

**Pull request merged**